src/Mvc/Mvc.Core/src/DependencyInjection/MvcCoreServiceCollectionExtensions.cs

```diff
@@ -65,7 +65,18 @@ public static IMvcCoreBuilder AddMvcCore(this IServiceCollection services)
 
         private static void ConfigureDefaultFeatureProviders(ApplicationPartManager manager)
         {
-            if (!manager.FeatureProviders.OfType<ControllerFeatureProvider>().Any())
+            var controllerFeaturePresent = false;
+
+            foreach(var feature in manager.FeatureProviders)
+            {
+                if(feature is ControllerFeatureProvider)
+                {
+                    controllerFeaturePresent = true;
+                    break;
+                }
+            }
+
+            if(!controllerFeaturePresent)
```

Comment:
You should be able to get exactly the same performance from this:

```suggestion
            if(!manager.FeatureProviders.Any(provider => provider is ControllerFeatureProvider))
```

Reply:
> exactly the same performance

It won't be "exactly" the same: every element would now involve a delegate invocation.

But I don't think that matters here.  I agree with Sam we should keep this simple and that this is a perfectly reasonable place to use Enumerable.Any.  I would change my tune if this were part of a default Blazor wasm app and was the last thing keeping Enumerable.Any alive in such an app and we could otherwise trim out Enumerable.Any, but it's not.